python/pyspark/sql/utils.py

```diff
@@ -24,6 +24,12 @@ class AnalysisException(Exception):
     """
 
 
+class IllegalArgumentException(Exception):
```

Comment:
i suspect this can be replaced with just the normal ValueError, but let's cc @davies to confirm.